### PR TITLE
chore: Migrate copy_dash endpoint to api v1

### PR DESCRIPTION
**Pull request by jfrag1**
### SUMMARY

This PR deprecates `superset/copy_dash/<dash_id>/` and introduces `api/v1/dashboard/<dash_id>/copy/` to replace it.

The new endpoint behaves nearly identically to the old one; the only functional differences are that the endpoint accepts a json body instead of a form, and it only responds with the data used by the frontend.

This PR also marks the `superset/save_dash/<dash_id>/` endpoint as deprecated as it is not called from client code anywhere.

### TESTING INSTRUCTIONS

[x] Add tests
[x] Test copying a dashboard from the frontend with "Save as"

### ADDITIONAL INFORMATION

- [ ] Has associated issue:
- [ ] Required feature flags:
- [ ] Changes UI
- [ ] Includes DB Migration (follow approval process in [SIP-59](https://github.com/apache/superset/issues/13351))
  - [ ] Migration is atomic, supports rollback & is backwards-compatible
  - [ ] Confirm DB migration upgrade and downgrade tested
  - [ ] Runtime estimates and downtime expectations provided
- [ ] Introduces new feature or API
- [ ] Removes existing feature or API

### CODE REVIEW

File: superset/dashboards/api.py [hunk 1]
```diff
@@ -1225,3 +1228,68 @@ def delete_embedded(self, dashboard: Dashboard) -> Response:
         for embedded in dashboard.embedded:
             DashboardDAO.delete(embedded)
         return self.response(200, message="OK")
+
+    @expose("/<id_or_slug>/copy/", methods=["POST"])
+    @protect()
```
Comment (dpgaspar) [inline superset/dashboards/api.py:1233]: let's make this endpoint permission be `can write on Dashboard` add copy_dash to https://github.com/apache/superset/blob/master/superset/constants.py#L151

File: superset/dashboards/dao.py [hunk 2]
```diff
@@ -299,3 +293,38 @@ def favorited_ids(dashboards: List[Dashboard]) -> List[FavStar]:
             )
             .all()
         ]
+
+    @classmethod
+    def copy_dashboard(
```
Comment (dpgaspar) [inline superset/dashboards/dao.py:298]: nit: how about just `def copy(` we know it's a dashboard from the class

**Review by betodealmeida — APPROVED** [review]
This looks good, but let's do what @dpgaspar recommended and add the method to `constants.py` so we don't add a new unneeded permission.

**Comment by jfrag1** [comment]
> This looks good, but let's do what @dpgaspar recommended and add the method to `constants.py` so we don't add a new unneeded permission.

I added it just now, but I'm curious if the `permission_name` decorator does the same thing?  Is the difference that with only the decorator, a DB record will still be created for the `copy_dash` permission but it won't do anything?

**Pull request merged**